https://bugs.gentoo.org/579928

From d84db34dbe39d55b4d7e868764c056689aa0793b Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Tue, 19 Apr 2016 01:56:41 -0400
Subject: [PATCH] extlinux: pull in sys/sysmacros.h for major/minor/makedev

These functions are defined in sys/sysmacros.h, so add the include to
main.c.  This is already handled correctly in mountinfo.c.  Otherwise
we get build failures like:

main.o: In function 'find_device_sysfs':
extlinux/main.c:1131: undefined reference to 'minor'

Upstream-Status: Pending
Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
Signed-off-by: Mike Frysinger <vapier@gentoo.org>

The first part wasn't enough in OE build, I had to add the same for syslinuxext.c.
---
 extlinux/main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/extlinux/main.c b/extlinux/main.c
index a7ebd49..ebff7ea 100644
--- a/extlinux/main.c
+++ b/extlinux/main.c
@@ -38,6 +38,7 @@
 #include <sysexits.h>
 #include <sys/ioctl.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 #include <sys/mount.h>
 #include <sys/vfs.h>
--- a/libinstaller/syslinuxext.c	2017-02-22 20:17:05.336869181 +0100
+++ b/libinstaller/syslinuxext.c	2017-02-22 20:16:47.500868751 +0100
@@ -2,6 +2,7 @@
 
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <getopt.h>
 #include <ext2fs/ext2fs.h>
 
